LicenseHeaderManager/LicenseHeadersPackage.cs

```diff
@@ -507,50 +536,22 @@ private void AddLicenseHeadersToAllFilesCallback (object sender, EventArgs e)
 
       HandleLinkedFilesAndShowMessageBox (addLicenseHeaderToAllFilesReturn.LinkedItems);
 
-      HandleAddLicenseHeaderToAllFilesReturn(obj, addLicenseHeaderToAllFilesReturn);
+      this.HandleAddLicenseHeaderToAllFilesInProjectReturn(obj, addLicenseHeaderToAllFilesReturn);
     }
 
-    private void HandleAddLicenseHeaderToAllFilesReturn(object obj,
+    private void HandleAddLicenseHeaderToAllFilesInProjectReturn(object obj,
       AddLicenseHeaderToAllFilesReturn addLicenseHeaderToAllFilesReturn)
     {
       var project = obj as Project;
       var projectItem = obj as ProjectItem;
       if (project == null && projectItem == null) return;
-      Project currentProject = project;
-
-      if (projectItem != null)
-      {
-        currentProject = projectItem.ContainingProject;
-      }
 
       if (addLicenseHeaderToAllFilesReturn.NoHeaderFound)
       {
-        //No license header found...
-        var page = (DefaultLicenseHeaderPage) GetDialogPage(typeof (DefaultLicenseHeaderPage));
-        var solutionSearcher = new AllSolutionProjectsSearcher();
-        var projects = solutionSearcher.GetAllProjects(_dte.Solution);
-
-        //If there is a licenseheader in the Solution 
-        if(projects.Any(projectInSolution => LicenseHeaderFinder.GetHeader(projectInSolution) != null))
+        // No license header found...
```

Comment:
Same argumentation as on the solution level. If another project contains a .licenseheader file, offer to link existing .licenseheader file (seems to have worked in contrary to the same action on solution level). If no project contains a .licenseheader file, we can offer the solution wide definition file.